sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -306,6 +306,15 @@ class SparkSqlParserSuite extends AnalysisTest {
       Project(UnresolvedAlias(concat) :: Nil, UnresolvedRelation(TableIdentifier("t"))))
   }
 
+  test("Prohibit binary comparisons chain") {
+    assertEqual("SELECT a = b",
+      Project(UnresolvedAlias(EqualTo(UnresolvedAttribute("a"), UnresolvedAttribute("b"))) :: Nil,
+        OneRowRelation()))
+
+    intercept("SELECT a = b = c", "Syntax error at or near")
```

Comment:
Nit: Let's add more cases to cover all the comparisions

Reply:
Ok.